### DISCUSSION TITLE
How to change the optimization method for the yaw angle optimization

### DISCUSSION THREAD

**Discussion by cjp606** [NatLabRockies/floris#390] [Q&A] [answered]
Hello everyone,

In the yaw angle simulation, the default optimization method is SLSQP. But if I want to use some other optimized algorithm such as the GA（Genetic Algorithm），PSO（Particle Swarm Optimization），ACO（Ant Clony Optimization）,etc. How can I to change the optimization method, can you give me a simple example about the code, due to I am not familar to the Python and FLORIS.

Thanks in advance.
Jiaping.Cui

**Comment by Bartdoekemeijer** [comment, marked as answer]
Hi Jiaping,

Which FLORIS version are you using? If you are using the most recent version, `v3.0`, we recommend that you use the "Serial Refine" optimization method. This optimization method is exemplified in the examples folder [here](https://github.com/NREL/floris/blob/main/examples/08_opt_yaw_single_ws.py). The Serial Refine method is recommend because of its computational efficiency, enabling entire wind rose optimizations in the matter of minutes. It's also shown to converge quickly to the optimal solution with much fewer function evaluations than the standardized optimization methods we tried.

If you insist on using SLSQP or one of the other global optimization methods, you should use the `SciPy` yaw optimizer class. Using SciPy for optimization is exemplified [here](https://github.com/NREL/floris/blob/df4622fab9547433f94ff13380456fa3360b73b1/examples/12_compare_yaw_optimizers.py#L47). This optimizer defaults to using the `SLSQP` method inside `SciPy.optimize`, but can use any optimization method provided with `SciPy`. You can assign the optimization method when loading the class. For example, if you would like to use Nelder-Mead, initialize the class as:

```
yaw_opt_scipy = YawOptimizationScipy(fi, opt_method="Nelder-Mead")
```
and everything else stays in same in terms of handling the optimizer class and processing the results.

**Reply by cjp606** [reply]
Dear Bart,

Thank you for your quickly reply. I will have a try according to your method. Thanks again!
